### PR TITLE
Remove imports to azcore package

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
No functional or public surface area changes.
Move some overlooked shared content to the shared package.
This will help with upcoming changes for generics.

- [ ] The purpose of this PR is explained in this or a referenced issue.
- [ ] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [ ] Tests are included and/or updated for code changes.
- [ ] Updates to [CHANGELOG.md][] are included.
- [ ] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go
[CHANGELOG.md]: https://github.com/Azure/azure-sdk-for-go/blob/main/CHANGELOG.md

### CODE REVIEW

File: sdk/azcore/arm/runtime/policy_bearer_token_test.go [hunk 1]
```diff
@@ -12,11 +12,11 @@ import (
 	"testing"
 	"time"
 
-	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore/arm"
 	armpolicy "github.com/Azure/azure-sdk-for-go/sdk/azcore/arm/policy"
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore/internal/pipeline"
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore/internal/shared"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
 	azpolicy "github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
```
Comment (seankane-msft) [inline sdk/azcore/arm/runtime/policy_bearer_token_test.go:20]: looks like the same thing is being imported twice, once with an alias

File: sdk/azcore/arm/runtime/policy_register_rp_test.go [hunk 2]
```diff
@@ -15,10 +15,11 @@ import (
 	"testing"
 	"time"
 
-	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
 	armpolicy "github.com/Azure/azure-sdk-for-go/sdk/azcore/arm/policy"
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore/internal/pipeline"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/internal/shared"
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore/log"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
```
Comment (seankane-msft) [inline sdk/azcore/arm/runtime/policy_register_rp_test.go:22]: same double import

**Review by seankane-msft — APPROVED** [review]
Two small nits on imports

File: sdk/azcore/arm/runtime/policy_bearer_token_test.go (hunk 1 above)
Comment (jhendrixMSFT) [reply sdk/azcore/arm/runtime/policy_bearer_token_test.go]: Good catch, fixed.

File: sdk/azcore/arm/runtime/policy_register_rp_test.go (hunk 2 above)
Comment (jhendrixMSFT) [reply sdk/azcore/arm/runtime/policy_register_rp_test.go]: Fixed

**Pull request merged**